test/integration/http_timeout_integration_test.cc

```diff
@@ -36,7 +36,7 @@ TEST_P(HttpTimeoutIntegrationTest, GlobalTimeout) {
 
   // Ensure we got a timeout downstream and canceled the upstream request.
   response->waitForHeaders();
-  ASSERT_TRUE(upstream_request_->waitForReset(std::chrono::milliseconds(0)));
+  ASSERT_TRUE(upstream_request_->waitForReset(std::chrono::milliseconds(10)));
```

Comment:
10ms is not very long on a loaded CI server. Why not just wait for a much longer period of time like 15s?

Reply:
sounds good to me, will change